examples/wp-rest-api/src/test-wp-rest-api.js

```diff
@@ -24,23 +25,23 @@ export function createApi( fetch = window.fetch ) {
 			},
 		}
 
-		static endpoints = {
-			posts: {
-				read: ( methods, endpointPath, params ) => {
-					const { get } = methods;
-					const fullEndpointPath = [ 'posts', ...endpointPath ];
-					const value = get( fullEndpointPath )( params );
-					return value;
-				},
+		static operations = {
+			read: ( { get } ) => ( resourceNames ) => {
+				const postPages = resourceNames.filter( name => startsWith( name, 'posts-page:' ) );
+
+				return postPages.reduce( ( requests, name ) => {
+					const params = JSON.parse( name.substr( name.indexOf( ':' ) + 1 ) );
```

Comment:
are you encoding a JSON string after the param?

also, for funnies…

```js
const [ type, data ] = name.split( ':' );
const params = JSON.parse( data );
```

Reply:
Ooh, I like that better, much more tidy! Totally going to use that.